### PR TITLE
feat: add devcontainer

### PR DESCRIPTION
**Pull request by sam32123**
## Current situation

Currently, this project has no .devcontainer in it's project, requiring every user to locally install all the dependencies.
## Fix

This should be changed to a .devcontainer, to make sure the local installation of dependencies always matches the production version of said dependencies.
Furthermore, this reduces the needfor local installation, apart from docker and a devcontainer supporting editor (for example: vscode)

## how to test:
See the readme :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,31 +10,35 @@ and contains things like upcoming activities and peculiar quotes.
...
+              If you do not have access to them, feel free to ask for them! However, you might not need them.
+              The secrets are only required to fetch content for the 'internal' pages, meant only for inside
+              the Sticky room: those visible only when you add `?internal=true` to the URL.
 
-      If, at the moment, you are not interested in those pages, you can just provide gibberish
-      secrets everywhere, and radio will work!
+              If, at the moment, you are not interested in those pages, you can just provide gibberish
+              secrets everywhere, and radio will work!
+
+        > :warning: **If you open the project in a dev container for the first time**: Initial installation may take a while. Docker is essentially creating a mini pc inside your pc, which can be up to 2gb of files to download. Please do not close your editor, but wait untill the left bottom pars of your editor states `Dev Container: Radio - SV Sticky`. It is then you can proceed on to step 4.
```
Comment (SilasPeters) [inline README.md:41]: Typo in the word 'pars', all the way at the end.

**Review by SilasPeters — DISMISSED** [review]
I have yet to test this, but I might as well share the following in the mean time. 

Just like in LaTeX, markdown files are supposed to have 1 sentence per line.
It makes it easier to comment on specific text as well.

**Comment by stickyPiston** [comment]
There exists a typescript dev container image with debian bookworn already, so we might just want to use that instead of using a custom dockerfile. The .devcontainer.json configuration would look like the following instead:
```json
{
	"name": "radio",
	"image": "mcr.microsoft.com/devcontainers/typescript-node:1-22-bookworm",
	"features": {
		"ghcr.io/devcontainers/features/github-cli:1": {
			"installDirectlyFromGitHubRelease": true,
			"version": "latest"
		}
	},
	"forwardPorts": [5173],
	"postCreateCommand": "npm i"
}
```

**Comment by sam32123** [comment]
> There exists a typescript dev container image with debian bookworn already, so we might just want to use that instead of using a custom dockerfile. The .devcontainer.json configuration would look like the following instead:
> 
> ```json
> {
> 	"name": "radio",
> 	"image": "mcr.microsoft.com/devcontainers/typescript-node:1-22-bookworm",
> 	"features": {
> 		"ghcr.io/devcontainers/features/github-cli:1": {
> 			"installDirectlyFromGitHubRelease": true,
> 			"version": "latest"
> 		}
> 	},
> 	"forwardPorts": [5173],
> 	"postCreateCommand": "npm i"
> }
> ```
> 
> Moreover, we need to adjust the vite.config.ts to host the development server in host mode. Without it, the development server will still run, but the browser outside of the container won't be able to connect to the server hosted in the container.
> 
> ```diff
> export default defineConfig({
>   plugins: [react()],
> +  server: {
> +    host: true,
> +    port: 5173
> +  }
> });
> ```

Is the latter part only the case for the .devcontainer you suggested, or also for the current? Because if you are also refering to the current, and I do not mean this rude, but you are incorrect. The host machine has acces to the server over [the](http://localhost:5173/) without adding the vite.config

File: README.md (hunk 1 above)
Comment (sam32123) [reply README.md]: done

**Comment by sam32123** [comment]
Oh boi it just keeps getting better. I think the 'official' image is a good idea, adding the postCreateCommand as well, and leaving out unnececary complexity as wel. Will update

**Review by stickyPiston — APPROVED** [review]
頑張ったね！

**Pull request merged**